[master] Porting #51701 to master

garethgreenaway opened the pull request:
Porting #51701 to master

geekinutah committed:
Log error when reactor tasks go to a full queue

Fixes #46431

geekinutah committed:
Adding space to log message as requested

SndR85 committed:
Add ARPCHECK option to rh7_eth template

Commit:
Use the minion's region if use_grain is enabled

This makes it possible for the master to be in a different region
than the minion and still fetch ec2 pillars. This requires the
use_grain configuration to be set to True and will fall back to the
master's region if there is no region grain.

Credit to John Nielsen for the code.

Signed-off-by: Carson Anderson <ca@carsonoid.net>

Commit:
Update acme.py

'/usr/local' is the path prefix on freebsd. this fixes:

```
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python2.7/site-packages/salt/state.py", line 1905, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python2.7/site-packages/salt/loader.py", line 1830, in wrapper
                  return f(*args, **kwargs)
                File "/usr/local/lib/python2.7/site-packages/salt/states/acme.py", line 108, in cert
                  group=group
                File "/usr/local/lib/python2.7/site-packages/salt/modules/acme.py", line 181, in cert
                  ret = {'comment': comment, 'not_after': expires(name)}
                File "/usr/local/lib/python2.7/site-packages/salt/modules/acme.py", line 253, in expires
                  return _expires(name).isoformat()
                File "/usr/local/lib/python2.7/site-packages/salt/modules/acme.py", line 67, in _expires
                  expiry = __salt__['tls.cert_info'](cert_file)['not_after']
                File "/usr/local/lib/python2.7/site-packages/salt/modules/tls.py", line 1595, in cert_info
                  with salt.utils.files.fopen(cert_path) as cert_file:
                File "/usr/local/lib/python2.7/site-packages/salt/utils/files.py", line 386, in fopen
                  f_handle = open(*args, **kwargs)  # pylint: disable=resource-leakage
              IOError: [Errno 2] No such file or directory: u'/etc/letsencrypt/live/somedomain/cert.pem'
```

backerman committed:
Add FreeBSD support to pf executor

terminalmage committed:
Add the saltcli tests to the filename map salt/client/*

This updates the filename map added in #47337 to ensure that the salt
CLI tests are run when anything in salt.client files are updated.

amendlik committed:
Add banner function to modules.slsutil

amendlik committed:
Add boolstr function to modules.slsutil

amendlik committed:
Use os.linesep for cross-platform compatibility

Commit:
Merge pull request #49981 from Lesvek/develop

Improved error logging for win_task.create_task_from_xml

garethgreenaway committed:
Copying changes from integration.sdb.test_vault over to integration.module.test_vault.

amendlik committed:
For the saltify cloud provider, use minion name as ssh_host

Commit:
Added `method_call` jinja filter.

Commit:
Added brief documentation and unit test.

Commit:
Change versionadded

Per @rallytime review

Commit:
added shutdown_host to vmware cloud

Commit:
added force recognition to reboot_host

Commit:
shutdown typo

Commit:
shutdown typo 2

Commit:
Add various ssh tests to the filemap for salt/utils/vt.py changes

Some of the salt-ssh tests use the salt/utils/vt.py functions. A recent
change in that file caused about 90 tests to start failing on the branch.

We need to make sure these tests run when changes to the vt.py util are
made in PRs.

mattp- committed:
new requisite form: onfail_all

when https://github.com/saltstack/salt/issues/22370 was introduced it
actually removed functionality, as sometimes it desirable to onfail if
ALL of the listed requisites are failure. This adds back an 'onfail_all'
form that allows for this behavior (it would make more sense to have
onfail default to OR logic, but that would break backcompat, so we add
a new form instead).

nick-wang committed:
drbd: support drbdadm to show status

drbd-overview is removed in the drbd-utils-9.6.0
Using 'drbdadm status' to show the state and
support multiple resources/peernodes/volumes.

nick-wang committed:
drbd: Add unit test case for drbd.status.

JD-Robertson committed:
Parse the command line from noaction operations and return the packages which would be modified

Signed-off-by: JD Robertson <jd.robertson@ni.com>

Review cleanup

Signed-off-by: JD Robertson <jd.robertson@ni.com>

More refactoring from review & fix pylint issues.

Signed-off-by: JD Robertson <jd.robertson@ni.com>

Fix comment.

Signed-off-by: JD Robertson <jd.robertson@ni.com>

Cleanup based on feedback

Signed-off-by: JD Robertson <jd.robertson@ni.com>

JD-Robertson committed:
Refactor for method length and fix multi-command cases

Signed-off-by: JD Robertson <jd.robertson@ni.com>

JD-Robertson committed:
Cleanup variable names.

Signed-off-by: JD Robertson <jd.robertson@ni.com>

JD-Robertson committed:
Update refactor to reduce cognitive complexity.

Signed-off-by: JD Robertson <jd.robertson@ni.com>

Commit:
Adding a keyvalue method to the file state module

For making changes to files based on a key/value pair.
This commit also includes addition of an integration test to test
the new behaviour added.

Commit:
fixing as many PyLint errors as possible

Commit:
mistook a blankline lint error for lack of blanklines between functions

Commit:
Lint error suggested salt.utils.fopen()

It should probably have suggested salt.utils.files.fopen() since
using salt.utils.fopen() breaks when running it through the
intergration test. It did seem to work when running it as a normal
state module though.

This should fix the additional lint errors.

Commit:
Trying to fix the windows2016-py2 build

Something on the windows2016-py2 build is throwing in what seems
like utf-16le encoding shenanigans in somewhere, but only for the
line actually being replaced, not any of the ones already there.
This obviously (and correctly I think) fails the test.

PermitRootLogin
becomes
P\x00e\x00r\x00m\x00i\x00t\x00R\x00o\x00o\x00t\x00L\x00o\x00g\x00i\x00n\x00

Also trying to be more OS friendly by using their native line
separators.

Commit:
Combined file.keyvalue_list and file.keyvalue

integration.states.test_file.FileTest.test_keyvalue was already
testing the file.keyvalue function since it used file.keyvalue_list
already.

Now there is only one function; file.keyvalue which takes either
2 parameters (key, and value) or a dictionary key_values to handle
multiple key values in a single file all at once.

Commit:
Fix linter and docstring errors

Commit:
fixing another linter warning

teizz committed:
Cleanup search_only parameter

Left over from when this function was still part of an execution
module. It got converted when it made a lot more sense to have it
in an actual state module

elisapa committed:
Fixed syntax error.

terminalmage committed:
Update client tests to reflect change in "no response" return

s0undt3ch committed:
Fix failing `integration.cli.test_grains.GrainsTargetingTest.test_grains_targeting_disconnected`

On the develop branch the last line is not what's being asserted against:
```
['disconnected:',
 '    Minion did not return. [No response]',
 '    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:',
 '    ',
 '    salt-run jobs.lookup_jid 20181206232209819226']
```
As opposed to what was expected on 2018.3.x:
```
['disconnected:',
 '    Minion did not return. [No response]']
```

Talkless committed:
Add advanced initdb options support in postgres_cluster.present

Some optional PostgreSQL database cluster options, such as data
checksuming mode, can only be set upon initialization.

Update state and execution modules to support these permanent options.

BrianSidebotham committed:
Allow clone_from setting in proxmox salt-cloud to be an integer as per documentation

dwoz committed:
Merge pull request #51095 from afischer-opentext-com/get_zone

Avoid reading Windows registry value which may not be in place

dwoz committed:
Merge pull request #51379 from arizvisa/win_system-module-set_domain_workgroup

Added an implementation of .set_domain_workgroup() that corresponds with .get_domain_workgroup()

OrangeDog committed:
Fix case for top-level binary shasums

For example:
b45991e49e50d89e8ceb1d6b7b0e881cece39e63 *filename

OrangeDog committed:
Update test_extract_hash to use a shasums file including binary hashes

OrangeDog committed:
Keep an example with one space

dbellavista committed:
Filter gluster CLI prompt (fixes #52318)

Commit:
Update events.rst

caught a typo

SMillerDev committed:
Fix slack state documentation for webhooks

amendlik committed:
Add missing period from slsutil.banner text

marbx committed:
Note on yappi

amendlik committed:
Add support for Javadoc-style banners in slsutil.banner

amendlik committed:
Add example documentation for slsutil.banner

amendlik committed:
Handle exception in slsutil.banner where width is too small to render

yohonet committed:
Update pillar.py

I believe something's missing in the documentation.

s0undt3ch committed:
Cleanup vault test

nikolas committed:
Fix typo in vault docs

atleast -> at least

dwoz committed:
Address beacon test flakiness

waynew committed:
Fix potential recursion/duplicate object detection

It's unlikely, but possible, that we could get a problem with over-eager
detection of recursion. Especially in the cases of short interned
strings or small numbers on CPython, since those are cached. Instead
we'll just check of it's a type that we're going to recurse on.

amendlik committed:
Update slsutil.banner example documentation

cbosdo committed:
Create virt.pool_defined state out of virt.pool_running

Users may want to use states to ensure a virtual storage pool is defined
and not enforce it to be running. Extract the code that performs the
pool definition / update from virt.pool_running state into a
virt.pool_defined.

Obviously the virt.pool_running state calls the virt.pool_defined one.
In such a case no additionnal test is needed for virt.pool_defined since
this is already tested with virt.pool_running.

cbosdo committed:
Add virt.update test parameter

In order to allow running dry-runs of virt.update module add a test
parameter. This will later be used by the virt states.

cbosdo committed:
Extract virt.defined state from virt.running

In order to ensure a virtual guest is defined independently of its
status, extract the corresponding code from the virt.running state.

This commit also handles the __opts__['test'] for the running state.
Since the update call only performs changes if needed, deprecate the
update parameter.

cbosdo committed:
Extract virt.network_defined from virt.network_running

Just like domains and storage pools, users may want to ensure a network
is defined without influencing it's status. Extract the code from
network_running state defining the network into a network_defined state.

While at it, support __opt__['test'] == True in these states. Updating
the network definition in the pool_defined state will come in a future
PR.

cbosdo committed:
Fix virt.update to handle None mem and cpu

virt.running state now may call virt.update with None mem and cpu
parameters. This was not handled in _gen_xml(). Also add some more tests
cases matching this for virt.update.

waynew commented on salt/modules/chocolatey.py:
```diff
@@ -930,8 +930,14 @@ def version(name, check_remote=False, source=None, pre_versions=False):
+            # in the dictionary that we return.
+            if pkg in available:
+                packages[pkg]['available'] = available[pkg]
+            continue
```
No need for implicit continue

waynew commented on salt/modules/chocolatey.py:
```diff
@@ -930,8 +930,14 @@ def version(name, check_remote=False, source=None, pre_versions=False):
-            packages[pkg] = {'installed': installed[pkg],
-                             'available': available[pkg]}
+            # Grab the current version from the package that was installed
+            packages[pkg] = {'installed': installed[pkg]}
```
I wonder - would this be better as `'available': available.get(pkg)`?

It produces an output like this:
```
local:
    ----------
    hosts:
        ----------
        available:
            None
        installed:
            - 1.1.0
```

Which feels better than:

```
local:
    ----------
    hosts:
        ----------
        installed:
            - 1.1.0
```

waynew reviewed (changes requested):
Definitely needs tests, could use some updates.

bdrung committed:
Allow skipping test cases that need Internet access

Mark test cases that need Internet access and allow skipping them by
setting the environment variable `NO_INTERNET`.

Signed-off-by: Benjamin Drung <benjamin.drung@cloud.ionos.com>

bdrung committed:
Move requires_network() restriction to BuildoutOnlineTestCase

`BuildoutOnlineTestCase.setUpClass` requires Internet connection.
Therefore move the `requires_network()` restriction to the class
definition to skip all test if no Internet connection is present.

Signed-off-by: Benjamin Drung <benjamin.drung@cloud.ionos.com>

s0undt3ch committed:
There's really no need to setup clear_func for each test.

This makes it so that all tests in this module take ~30 seconds to run
as opposed to ~20s for each test

Commit:
states/svn: set ret[comment] appropriately for 'svn export'

A value "out" is computed but never used inside export(). Per similar
code in latest() (ie 'svn update'), this "out" should be assigned to
ret[comment].

Signed-off-by: J Rick Ramstetter <rick@anteaterllc.com>

s0undt3ch committed:
This was a bad idea caught soon enough.

s0undt3ch committed:
Some improvements on Salt's global exception handler

s0undt3ch committed:
Better handling of `close()` when called from `__del__`

```
Exception AttributeError: "'AsyncReqMessageClient' object has no attribute '_closing'" in <bound method AsyncReqMessageClient.__del__ of <salt.transport.zeromq.AsyncReqMessageClient object at 0x1a0e82e90>> ignored
```

s0undt3ch committed:
Don't fail on windows

s0undt3ch committed:
Rework test group selection logic

s0undt3ch committed:
No module level globals

twangboy committed:
Fix windows lgpo tests

terminalmage committed:
black

terminalmage committed:
lint

s0undt3ch committed:
Fix wrong/bad/buggy waiting logic

dwoz committed:
Merge pull request #55862 from Ch3LL/pr_req

Write docs for SEP 10 - PR Merge Requirements

dwoz committed:
Merge pull request #56044 from terminalmage/missing-shadow-entry

shadow.set_password: handle case where user has no shadow entry

dwoz committed:
Merge pull request #56227 from bdrung/fix-symlink-test

Fix broken symlink in test_symlink_list

sabtom committed:
Merge remote-tracking branch 'upstream/master' into bugfix/py3_tomcat

sabtom committed:
Merge remote-tracking branch 'upstream/master' into bugfix/py3_tomcat

Ch3LL committed:
Revert "add towncrier to manage changelog"

This reverts commit 57d160cc606f405f7929cde2bc7dde17e575a1c4.

Ch3LL committed:
Revert "update pydir call and fix format"

This reverts commit 7468883dc005e949e9304937747a0f256469c184.

Commit:
Merge branch 'master' into its

twangboy committed:
Backport #53994

Fix typo perkissive to permissive

twangboy committed:
Update changelog

dwoz committed:
Merge pull request #56719 from xiaohuazi123/Fix-prompt-some-functions-mssql.py

Fix prompt of some functions in module mssql.py

dwoz committed:
Merge pull request #56728 from Ch3LL/town_change

Revert #56641

s0undt3ch committed:
Make sure everything is sync'ed on Windows.

On Windows 2016 we seem to be getting:
```
base:issue-1876' failed: Jinja variable 'salt.utils.templates.AliasedLoader object' has no attribute 'runtests_helpers.get_salt_temp_dir_for_path'
```
This commit tries to resolve that

s0undt3ch committed:
Remove unused functions

dwoz committed:
Merge pull request #56249 from Chupaka/its

Replace incorrectly used "it's" by "its"

twangboy committed:
Update changelog

twangboy committed:
Fix black

twangboy committed:
Fix more black

twangboy committed:
Final black

twangboy committed:
Very final black

twangboy committed:
Add changelog

terminalmage committed:
add test case

terminalmage committed:
Add test case

Commit:
fix cloud tests

terminalmage committed:
Add test case

terminalmage committed:
Simplify mock

boltronics committed:
Fix a typo in the intro_scale tutorial

nicoflex1 committed:
Fix config and doc for option master_roots

s0undt3ch committed:
Improve tests setup/teardown.

Additionally, don't use a fixed path to the `vaul` binary

s0undt3ch committed:
Remove Py2 specific tests. Improve test class setup/teardown.

s0undt3ch committed:
Fix the pylint and codeclimate.

s0undt3ch committed:
Fix: divided into small funcs for Cognitive Complexity.

s0undt3ch committed:
Add docs for all functions.

s0undt3ch committed:
No test module global variables.

s0undt3ch committed:
Lint and formatting fixes. Update `versionadded` to Sodium.

waynew committed:
Skip slowest tests

dwoz committed:
Merge pull request #56774 from waynew/runtests-skip-slow

Skip slowest tests for PR port jam

dhiltonp committed:
generify masterminion returners function not found error - may be because returner isn't loaded

mattp- committed:
silence some high-noise log messages in pillar cache

on larger installations these tend to fill all of the logs. lets drop
the level of them since they aren't very useful.

dwoz committed:
Port #52560 to master

Commit:
Port #48990 to master

Fix static configuration.
If request mode is static, then all settings will be in INI_FILE.

Fix test_static_all.
On older nilrt, just one static dns is supported.

pengyao committed:
Async ping_all_connected_minions

twangboy committed:
Update changelog

waynew committed:
Skip all docker tests due to teardown - revert with #56774

dwoz committed:
Merge pull request #56796 from waynew/fix-docker-slow-skip

Skip all docker tests due to teardown - revert with #56774

dwoz committed:
Merge pull request #56784 from bryceml/53324

Port #53324 to master

dwoz committed:
Merge pull request #56780 from bryceml/53571

Port #53571 into master

dwoz committed:
Merge pull request #56776 from bryceml/53964

Port #53964 into master

dwoz committed:
Merge pull request #56756 from boltronics/tutorial_typo_fix

Fix a typo in the intro_scale tutorial

dwoz committed:
Merge pull request #56781 from bryceml/53481

Port #53481 to master

dwoz committed:
Merge branch 'master' into master-port-49129

dwoz committed:
Merge branch 'master' into master-port-49097

dwoz committed:
Merge branch 'master' into master-port-49094

dwoz committed:
Merge branch 'master' into master-port-51670

dwoz committed:
Merge branch 'master' into master-port-50998

dwoz committed:
Merge branch 'master' into master-port-48775

dwoz committed:
Merge branch 'master' into 52763

dwoz committed:
Merge branch 'master' into master-port-52560

dwoz committed:
Merge branch 'master' into 52546

dwoz committed:
Merge branch 'master' into backport_51379

dwoz committed:
Merge branch 'master' into 53622

dwoz committed:
Merge branch 'master' into port_54018

dwoz committed:
Merge branch 'master' into port_54016

dwoz committed:
Merge branch 'master' into port-to-master/50627

dwoz committed:
Merge branch 'master' into port-to-master/50410

dwoz committed:
Merge branch 'master' into port-to-master/50230

dwoz committed:
Merge branch 'master' into port-to-master/50779

dwoz committed:
Merge branch 'master' into port-to-master/50152

dwoz committed:
Merge pull request #56762 from nee2c/fix-master-roots-master

Fix master roots master

dwoz committed:
Merge pull request #56760 from s0undt3ch/port-to-master/49216

Port #49216 to master

dwoz committed:
Merge branch 'master' into port-to-master/49146

dwoz committed:
Merge pull request #56755 from saltstack/fix_digital_ocean_cloud_tests

fix cloud tests

dwoz committed:
Merge pull request #56806 from techhat/master-port-49129

Port #49129 to master

dwoz committed:
Merge pull request #56797 from garethgreenaway/master-port-50998

[master] Port #50998 to master

dwoz committed:
Merge pull request #56787 from techhat/master-port-48775

Port #48775 to master

dwoz committed:
Merge pull request #56757 from ut3/rr-svn-py-comment

states/svn: set ret[comment] appropriately for 'svn export'

dwoz committed:
Merge pull request #56789 from bryceml/52763

Port #52763 to master

dwoz committed:
Merge pull request #56791 from bryceml/52546

Port #52546 to master

dwoz committed:
Merge pull request #56790 from xeacott/master-port-52560

Port #52560 to master

dwoz committed:
Merge pull request #56792 from twangboy/backport_51379

Backport #51379

dwoz committed:
Merge pull request #56747 from s0undt3ch/hotfix/pytest-patch-15

[PyTest #33 - Extract] This was a bad idea caught soon enough.

dwoz committed:
Merge pull request #56733 from s0undt3ch/hotfix/pytest-patch-1

[PyTest #33 - Extract] Code cleanup + Make sure Windows is sync'ed

dwoz committed:
Merge branch 'master' into 52854

dwoz committed:
Merge branch 'master' into master-port-50704

dwoz committed:
Blacken changed files

dwoz committed:
Merge branch 'master' into master-port-49097

dwoz committed:
Merge pull request #56778 from bryceml/53622

Port #53628 into master

dwoz committed:
Merge pull request #56802 from techhat/master-port-49097

Port #49097 to master

dwoz committed:
Merge pull request #56786 from bryceml/52854

Port #52854 to master

dwoz committed:
Merge pull request #56793 from garethgreenaway/master-port-50704

[master] Port #50704 to master

dwoz committed:
Merge pull request #56766 from s0undt3ch/port-to-master/50779

Port #50779 to master

dwoz committed:
Merge pull request #56768 from s0undt3ch/port-to-master/50410

Port #50410 to master

dwoz committed:
Merge pull request #56767 from s0undt3ch/port-to-master/50230

Port #50230 to master

dwoz committed:
Merge pull request #56770 from s0undt3ch/port-to-master/50627

Port #50627 to master

dwoz committed:
Merge pull request #56777 from Ch3LL/port_54018

Port #54018 into master

dwoz committed:
Merge pull request #56775 from Ch3LL/port_54016

Port #54016 to master

dwoz committed:
Merge pull request #56759 from s0undt3ch/port-to-master/49146

Port #49146 to master

dwoz committed:
Merge branch 'master' into 52324

dwoz committed:
Merge pull request #56801 from techhat/master-port-49094

Port #49094 to master

dwoz committed:
Merge pull request #56794 from bryceml/52324

Port #52324 to master

dwoz committed:
Merge branch 'master' into port-to-master/49396

dwoz committed:
Merge branch 'master' into port-to-master/50516

dwoz committed:
Merge pull request #56799 from garethgreenaway/master-port-51670

[master] Port #51670 to master

dwoz committed:
Merge branch 'master' into backport_49981

dwoz committed:
Merge pull request #56753 from twangboy/backport_51095

Backport #51095

dwoz committed:
Merge pull request #56769 from s0undt3ch/port-to-master/50516

Port #50516 to master

dwoz committed:
Merge pull request #56761 from s0undt3ch/port-to-master/49396

Port #49396 to master

dwoz committed:
Merge branch 'master' into backport_53994

dwoz committed:
Merge pull request #56763 from s0undt3ch/port-to-master/50102

Port #50102 to master

dwoz committed:
Merge pull request #56712 from terminalmage/bp-50135

Port #50135 to master

dwoz committed:
Merge branch 'master' into bp-50177

dwoz committed:
Merge branch 'master' into bp-51002

dwoz committed:
Merge branch 'master' into bp-51013

dwoz committed:
Merge pull request #56697 from terminalmage/bp-51253

Port #51253 to master

dwoz committed:
Merge pull request #56695 from terminalmage/bp-51923

Port #51923, #50591, #50661, #50694 to master

dwoz committed:
Merge pull request #56694 from terminalmage/bp-51470

Port #51470 to master

dwoz committed:
Merge pull request #56765 from s0undt3ch/port-to-master/50152

Port #50152 to master

dwoz committed:
Merge pull request #56731 from twangboy/backport_53994

Backport #53994

dwoz committed:
Merge branch 'master' into backport_49981

dwoz committed:
Merge pull request #56751 from twangboy/backport_49981

Backport 49981

dwoz committed:
Merge pull request #56711 from terminalmage/bp-50177

Port #50177 to master

dwoz committed:
Merge pull request #56707 from terminalmage/bp-51002

Port #51002 to master

dwoz committed:
Merge branch 'master' into bp-51013

dwoz committed:
Merge pull request #56705 from terminalmage/bp-51013

Port #51013 to master

s0undt3ch committed:
Concentrate random string generation into a generic helper

dwoz committed:
Merge pull request #56746 from s0undt3ch/hotfix/pytest-patch-14

[PyTest #33 - Extract] Some improvements on Salt's global exception handler

dwoz committed:
Merge pull request #56809 from joechainz/master-port-53862

Cleanup vault test

dwoz committed:
Merge pull request #56745 from s0undt3ch/hotfix/pytest-patch-13

[PyTest #33 - Extract] Better handling of `close()` when called from `__del__`

dwoz committed:
Merge pull request #56734 from s0undt3ch/hotfix/pytest-patch-2

[PyTest #33 - Extract] Fix wrong/bad/buggy waiting logic

DmitryKuzmenko committed:
Merge branch 'master' into master-port-48990

cbosdo committed:
Merge branch 'master' into virt-defined-states

dwoz committed:
Merge pull request #56795 from DSRCorporation/master-port-48990

Port #48990 to master

dwoz committed:
Merge pull request #56735 from s0undt3ch/hotfix/pytest-patch-3

[PyTest #33 - Extract] Concentrate random string generation into a generic helper

dwoz committed:
Merge pull request #56736 from s0undt3ch/hotfix/pytest-patch-4

[PyTest #33 - Extract] Fix windows lgpo tests

dwoz committed:
Merge branch 'master' into hotfix/pytest-patch-5

dwoz committed:
Merge pull request #56737 from s0undt3ch/hotfix/pytest-patch-5

[PyTest #33 - Extract] No module level globals

dwoz committed:
Merge pull request #56738 from s0undt3ch/hotfix/pytest-patch-6

[PyTest #33 - Extract] Rework test group selection logic

dwoz committed:
Merge pull request #56740 from s0undt3ch/hotfix/pytest-patch-8

[PyTest #33 - Extract] Don't fail on windows

dwoz committed:
Merge pull request #56741 from s0undt3ch/hotfix/pytest-patch-9

[PyTest #33 - Extract] There's really no need to setup clear_func for each test.

dwoz committed:
Merge pull request #56681 from terminalmage/issue56654

Implicitly ignore epoch when no epoch in desired version, and ignore_epoch not passed

Commit:
rm file

mattp- committed:
Port #53517 to master

pbeetham committed:
Port #52138 to master

Akm0d committed:
Port #53537 to master

alexey-zhukovin committed:
Merge branch 'master' into master-port-52138

Ch3LL committed:
Fix lint test_pkg - readd skipif

dwoz committed:
Merge pull request #52699 from aplanas/fix_brtfs

btrfs: Add properties state

dwoz committed:
Merge pull request #53100 from terminalmage/ip_networks

salt.modules.network: Add functions to get minion's networks

dwoz committed:
Merge pull request #54468 from sathieu/lazy_envs

Lazily gather the lists of available sls data from the master

dwoz committed:
Merge pull request #55963 from bdrung/master-fix-doc-import-kubernetesmod

Mock kubernetes when building the documentation

Akm0d committed:
Fix tests

dwoz committed:
Merge pull request #56270 from lythobv/bugfix/py3_tomcat

Force bytestrings to string in tomcat module

mattp- committed:
Fix tests

dwoz committed:
Merge pull request #56840 from Ch3LL/fix_lint

Fix lint test_pkg - readd skipif

dwoz committed:
Merge branch 'master' into master-port-52138

dwoz committed:
Merge pull request #55814 from cbosdo/virt-defined-states

Add virt.*defined states

dwoz committed:
Merge branch 'master' into master-port-53537

dwoz committed:
Merge branch 'master' into master-port-53517

dwoz committed:
Merge pull request #56831 from joechainz/master-port-50264

Master port 50264

dwoz committed:
Merge pull request #56838 from alexey-zhukovin/master-port-52138

Port #52138 to master

dwoz committed:
Merge pull request #56835 from xeacott/master-port-53537

Port #53537 to master

dwoz committed:
Merge pull request #56832 from xeacott/master-port-53517

Port #53517 to master

dwoz committed:
Merge pull request #56101 from mchugh19/multi-assert

Saltcheck allow multiple asserts against module output

dwoz committed:
Merge pull request #56278 from mcalmer/fix-zypper-option-abbr

use full option name instead of undocumented abbreviation

dwoz committed:
Merge pull request #56280 from bdrung/internet-access

Allow skipping test cases that need Internet access

dwoz committed:
Merge pull request #56298 from raddessi/master.state-test

Added test function to state module

dwoz committed:
Merge pull request #56408 from lorengordon/lockoutduration

Uses correct zero value for lgpo LockoutDuration

garethgreenaway committed:
Porting PR #51701 to 2019.2.1

garethgreenaway committed:
Adding test for chocolately.version changes.

garethgreenaway committed:
Merge branch '2019_2_1_port_51701' of github.com:garethgreenaway/salt into 2019_2_1_port_51701

garethgreenaway commented:
Replaced with https://github.com/saltstack/salt/pull/56870
